napari_tiff/napari_tiff_metadata.py

```diff
@@ -0,0 +1,395 @@
+        contrast_limits=contrast_limits,
+        blending=blending,
+        visible=visible,
+    )
+    return kwargs
+
+
+def get_imagej_metadata(tif: TiffFile) -> dict[str, Any]:
+    """Return napari LayerData from ImageJ hyperstack."""
+    # TODO: ROI overlays
+    ijmeta = tif.imagej_metadata
+    series = tif.series[0]
+
+    dtype = series.dtype
+    axes = series.axes
+    shape = series.shape
+    page = series.pages[0]
+    rgb = page.photometric == 2 and shape[-1] in (3, 4)
+    mode = ijmeta.get("mode", None)
+    channels = ijmeta.get("channels", 1)
+    channel_axis = None
+
+    name = None
+    scale = None
+    colormap = None
+    contrast_limits = None
+    blending = None
+    visible = True
+
+    if mode in ("composite", "color"):
+        channel_axis = axes.find("C")
+        if channel_axis < 0:
+            channel_axis = None
+
+    if channel_axis is not None:
+        channels = shape[channel_axis]
+        channel_only = channels == ijmeta.get("images", 0)
+
+        if "LUTs" in ijmeta:
+            colormap = [Colormap(c.T / 255.0) for c in ijmeta["LUTs"]]
+        elif mode == "grayscale":
+            colormap = "gray"
+        elif channels < 8:
+            colormap = ["red", "green", "blue", "gray", "cyan", "magenta", "yellow"][
+                :channels
+            ]
+
+        if "Ranges" in ijmeta:
+            contrast_limits = numpy.array(ijmeta["Ranges"]).reshape(-1, 2)
+            contrast_limits = contrast_limits.tolist()
+
+        if channel_only and "Labels" in ijmeta:
+            name = ijmeta["Labels"]
+        elif channels > 1:
+            name = [f"Channel {i}" for i in range(channels)]
+
+        if mode in ("color", "grayscale"):
+            blending = "opaque"
+
+    elif axes[-1] == "S" and dtype == "uint16":
+        # RGB >8-bit
+        channel_axis = axes.find("S")
+        if channel_axis >= 0 and shape[channel_axis] in (3, 4):
+            rgb = False
+            n = shape[channel_axis]
+            visible = [True, True, True]
+            colormap = rgb_colormaps(samples=4)[:n]
+            name = ["Red", "Green", "Blue", "Alpha"][:n]
+            blending = ["additive", "additive", "additive", "translucent"][:n]
+        else:
+            channel_axis = None
+
+    scale = {}
+    res = page.tags.get("XResolution")
+    if res is not None:
+        scale["X"] = res.value[1] / max(res.value[0], 1)
+    res = page.tags.get("YResolution")
+    if res is not None:
+        scale["Y"] = res.value[1] / max(res.value[0], 1)
+    scale["Z"] = abs(ijmeta.get("spacing", 1.0))
+    if channel_axis is None:
+        scale = tuple(scale.get(x, 1.0) for x in axes if x != "S")
+    else:
+        scale = tuple(scale.get(x, 1.0) for x in axes if x not in "CS")
+
+    kwargs = dict(
+        rgb=rgb,
+        channel_axis=channel_axis,
+        name=name,
+        scale=scale,
+        colormap=colormap,
+        contrast_limits=contrast_limits,
+        blending=blending,
```

Comment:
I am not convinced that we should provide blending here. I think that this should be handled on viewer side.